core/src/main/scala/akka/kafka/internal/TransactionalProducerStage.scala

```diff
@@ -159,21 +190,35 @@ private final class TransactionalProducerStageLogic[K, V, P](stage: Transactiona
   override def onCompletionFailure(ex: Throwable): Unit = {
     log.debug("Aborting transaction due to stage failure")
     abortTransaction()
+    batchOffsets.committingFailed()
     super.onCompletionFailure(ex)
   }
 
   private def commitTransaction(batch: NonemptyTransactionBatch, beginNewTransaction: Boolean): Unit = {
     val group = batch.group
     log.debug("Committing transaction for consumer group '{}' with offsets: {}", group, batch.offsetMap())
```

Comment:
[Minor] Use the local variable offsetMap so we don't have to to build for each log line?